packages/plugin-dev/src/node/hosted-plugins-manager.ts

```diff
@@ -90,27 +91,27 @@ export class HostedPluginsManagerImpl implements HostedPluginsManager {
     }
 
     isWatchCompilationRunning(uri: string): Promise<boolean> {
-        const pluginPath = this.getFsPath(uri);
+        const pluginPath = FileUri.fsPath(uri);
 
         return new Promise(resolve => resolve(this.watchCompilationRegistry.has(pluginPath)));
     }
 
-    protected runWatchScript(path: string): Promise<void> {
-        const watchProcess = cp.spawn('yarn', ['run', 'watch'], { cwd: path });
-        watchProcess.on('exit', () => this.unregisterWatchScript(path));
+    protected runWatchScript(pluginRootPath: string): Promise<void> {
+        const watchProcess = cp.spawn('yarn', ['run', 'watch'], { cwd: pluginRootPath, shell: true });
+        watchProcess.on('exit', () => this.unregisterWatchScript(pluginRootPath));
 
-        this.watchCompilationRegistry.set(path, watchProcess);
+        this.watchCompilationRegistry.set(pluginRootPath, watchProcess);
         this.hostedPluginSupport.sendLog({
-            data: 'Compilation watcher has been started in ' + path,
+            data: 'Compilation watcher has been started in ' + pluginRootPath,
             type: LogType.Info
         });
         return Promise.resolve();
     }
 
-    protected unregisterWatchScript(path: string): void {
-        this.watchCompilationRegistry.delete(path);
+    protected unregisterWatchScript(pluginRootPath: string): void {
```

Comment:
Hello, why do we need to rename path to pluginRootPath, it's adding extra changes ?

Reply:
I think it improves the code, as it better describes the purpose of the variable and puts it more in line with the rest of the module.

Reply:
@westbury Was there a compilation warning/error because of a shadowed module variable `path`?

Reply:
@akosyakov No, the rename was not technically required.  I was just helping to make the code a little bit clearer by using the same name for paths that are specifically a path to the plugin root.